src/puppetlabs/ring_middleware/core.clj

```diff
@@ -46,6 +107,79 @@
   [handler]
   (fn [{:keys [ssl-client-cert] :as req}]
     (let [cn  (some-> ssl-client-cert
-                      get-cn-from-x509-certificate)
+                      ssl-utils/get-cn-from-x509-certificate)
           req (assoc req :ssl-client-cn cn)]
       (handler req))))
+
+(schema/defn ^:always-validate wrap-data-errors
+  "A ring middleware that catches slingshot errors of :type
+  :request-dava-invalid and returns a 400."
+  ([handler :- IFn]
+   (wrap-data-errors handler :json))
+  ([handler :- IFn
+    type :- ResponseType]
+   (let [code 400
+         response (fn [e]
+                    (log/error "Submitted data is invalid: "  (:message e))
+                    (case type
+                      :json (json-response code {:error e})
+                      :plain (plain-response code (:message e))))]
+     (fn [request]
+       (try+ (handler request)
+             (catch
+               #(contains? #{:request-data-invalid
+                             :user-data-invalid
+                             :service-status-version-not-found}
+                           (:type %))
+               e
+               (response e)))))))
```

Comment:
This implementation is more complicated than what was originally in `wrap-user-data-errors` in file sync, so I'm assuming this was combined with something else that needed to return a plaintext response?